### PR TITLE
Don't steal reference from existing Python object

### LINKED ISSUE #551: Passing PyObject through include/eval turns it into PyNULL (which causes segfaults)
```julia
julia> using PyCall

julia> obj = pyimport("sys")  # get some PyObject
PyObject <module 'sys' (built-in)>

julia> py"""
       ns = {}
       def set(name):
           ns[name] = $include_string($Main, name)
       """

julia> py"set"("obj")

julia> obj
PyObject NULL

julia> py"lambda x: x"(obj)

signal (11): Segmentation fault
```

There are two problems: (1) `py"set"("obj")` turns `obj` to `PyNULL` (ref count issue?) and (2) passing `PyNULL` to Python function is not guarded enough.

Note that `py"set"` is modeling what we do in PyJulia all the time so it's not a contrived example.

### PR DESCRIPTION
**Pull request by tkf**
closes #551

I'm not very familiar with Python C API but I think this fixes the bug.

### CODE REVIEW

**Comment by stevengj** [comment]
Thanks, good catch!

**Comment by stevengj** [comment]
Can you audit the other usages of `pystealref!` in case of similar problems?

It seems like it would be cleaner to define:
```jl
pyptr(x) = pystealref!(PyObject(x))
pyptr(x::PyObject) = pyincref_(x.o)
```
and then call `pyptr(x)` in all the cases where we would have previously called `pystealref!(PyObject(x))`

**Comment by tkf** [comment]
I grepped `pystealref!` and found two more similar bugs which are fixed in the same way.

File: src/PyCall.jl [hunk 1]
```diff
@@ -133,6 +133,16 @@ function pystealref!(o::PyObject)
...
 
+"""
+    pyreturn(x) :: PyPtr
+
+Prepare `PyPtr` from `x` for passing it to Python.  If `x` is already
+a `PyObject`, the refcount is incremented.  Otherwise a `PyObject`
+wrapping/converted from `x` is created.
+"""
+pyreturn(x::Any) = pystealref!(PyObject(x))
+pyreturn(x::PyObject) = pyincref_(x.o)
```
Comment (tkf) [inline src/PyCall.jl]: I named it `pyreturn` since I thought that's more intuitive.  Of course, it is super easy to replace it to `pyptr`.

**Comment by stevengj** [comment]
Thanks!

**Pull request merged**